command/agent/config.go

```diff
@@ -444,6 +444,9 @@ type ServerConfig struct {
 	// ServerJoin contains information that is used to attempt to join servers
 	ServerJoin *ServerJoin `hcl:"server_join"`
 
+	// SystemSchedulerPreemptionEnabledDefault is used to determin whether to enable system preemption by default in a new cluster
+	SystemSchedulerPreemptionEnabledDefault *bool `hcl:"system_scheduler_preemption_enabled_default"`
```

Comment:
What if we added a `Defaults` struct for this field so we can reuse it for all API configured settings? I think we should provide file-configurable-defaults for all scheduler configuration parameters to better support Packer & TF users who don't want to have to run some commands to tweak their cluster post deployment.

Reply:
Ha, just noticed your comment above saying almost the exact same thing @notnoop. :sweat_smile: 

Another naming idea based off of yours:

```hcl
server {
  enabled = true

 # ... a bunch of other stuff ...

  scheduler_defaults {
    system_job_preemption = false
  }
}
```

We can drop `_enabled` as other bool fields don't have it. Otherwise I think some sort of qualifier is necessary for "system" as shortening it to `system_preemption` may not be immediately obvious to users that its referring to system _jobs_ since the word "system" is quite generic.

Reply:
I agree: a generic defaults make sense.  Though, I would prefer to use the same json format or key names that the API accepts.  Given that the API is the canonical way for configuration, I opted to use the same API terms as-is.  Making them hcl-friendly risks hcl format being behind/divergent from the api format and complicates documentation/implementation.

Reply:
Ah right, we should probably follow the jobspec's lead here:

1. The *style* should vary by format (underscores for HCL, Caps for JSON)
2. Otherwise the *names* should match